src/main/java/htsjdk/samtools/reference/AbstractFastaSequenceFile.java

```diff
@@ -85,29 +75,41 @@
         this.sequenceDictionary = sequenceDictionary;
     }
 
-    protected static File findSequenceDictionary(final File file) {
-        final Path dictionary = findSequenceDictionary(IOUtil.toPath(file));
-        if (dictionary == null) {
+    /** Attempts to find and load the sequence dictionary if present. */
+    protected SAMSequenceDictionary findAndLoadSequenceDictionary(final Path fasta) {
+        final Path dictPath = findSequenceDictionary(path);
+        if (dictPath == null) {
             return null;
         }
-        return dictionary.toFile();
+        else {
+            IOUtil.assertFileIsReadable(dictPath);
+            try (InputStream dictionaryIn = IOUtil.openFileForReading(dictPath)) {
+                return ReferenceSequenceFileFactory.loadDictionary(dictionaryIn);
+            }
+            catch (Exception e) {
```

Comment:
`catch` should be on previous line

Reply:
Probably?  I was just moving this code around, not actually changing it.